src/main/resources/templates/index.html

```diff
@@ -6,44 +6,179 @@
+        }
+
+        function handleChangeIncludeEventType() {
+            let selected = Array.from(document.getElementsByClassName("include-event-checkbox"))
+            let checkAll = !selected.some((el) => el.checked)
+            document.getElementById("include-all-event-types").checked = checkAll
+            document.getElementById("include-all-event-types").disabled = checkAll
+            if (!checkAll) excludeNoEventTypes()
+        }
+
+        function excludeNoEventTypes() {
+            document.getElementById("exclude-no-event-types").checked = true
+            document.getElementById("exclude-no-event-types").disabled = true
+            Array.from(document.getElementsByClassName("exclude-event-checkbox")).map((element) => element.checked = false)
+        }
+
+        function handleChangeExcludeEventType() {
+            let selected = Array.from(document.getElementsByClassName("exclude-event-checkbox"))
+            let checkAll = !selected.some((el) => el.checked)
+            document.getElementById("exclude-no-event-types").checked = checkAll
+            document.getElementById("exclude-no-event-types").disabled = checkAll
+            if (!checkAll) includeAllEventTypes()
+        }
     </script>
 </head>
 <body>
 <div layout:fragment="content" class="govuk-grid-column-two-thirds">
     <h1 class="govuk-heading-xl">Offender Events UI</h1>
-
     <div>
+
+        <!-- Filter form -->
         <form id="filter-form" class="govuk-form-group" action="#" th:action="@{/messages}">
-            <table class="govuk-table">
-                <thead class="govuk-table__head">
-                    <tr class="govuk-table__row">
-                        <th class="govuk-table__header" scope="col">Event Type</th>
-                        <th class="govuk-table__header" scope="col" style="colspan:2">Text</th>
-                    </tr>
-                </thead>
-                <tbody class="govuk-table__body">
-                    <tr class="govuk-table__row">
-                        <td class="govuk-table__cell">
-                            <select id="event-type-filter" name="event-type-filter" class="govuk-select">
-                                <option value="">All</option>
-                                <option th:each="eventType : ${allEventTypes}" th:value="${eventType}" th:text="${eventType}" th:selected="${eventType == eventTypeFilter}"/>
-                            </select>
-                        </td>
-                        <td class="govuk-table__cell" style="colspan:2">
-                            <input id="text-filter" name="text-filter" class="govuk-input" th:value="${textFilter}"/>
-                        </td>
-                    </tr>
-                </tbody>
-            </table>
-            <input class="govuk-button" type="submit" value="Filter" />
+            <div class="govuk-accordion" data-module="govuk-accordion">
```

Comment:
Ended up with an accordion for this because by default I want to hide the event type filters - it builds the filter dynamically based upon what event types are available and this could potentially eat a lot of screen space.

Reply:
As a side note, if you were to check out the project and play with the filter you'd notice that there are some alignment issues that aren't very pretty.  As this is just a tool for devs I've put most effort into making this useful but not a lot into making it pretty.  It's kind of ok.